samza-core/src/main/java/org/apache/samza/application/StreamApplicationDescriptorImpl.java

```diff
@@ -286,8 +164,8 @@ public String getNextOpId(OpCode opCode, String userDefinedId) {
     }
 
     String nextOpId = String.format("%s-%s-%s-%s",
-        config.get(JobConfig.JOB_NAME()),
-        config.get(JobConfig.JOB_ID(), "1"),
+        getConfig().get(JobConfig.JOB_NAME()),
```

Comment:
Since you are changing here: can these use JobConfig.getName and JobConfig.getId?